File: frontend/src/modelos.ts
export interface WebApp {
  podcast: Podcast;
  descargas: TotalDescargas;
  descargando: boolean;
  numeroDescargas: number;
}

interface Podcast {
  error_critico?: string;
  errores: string[];
  info: Info;
  programas: Programas[];
}

interface Info {
  nombre?: string;
  descripcion?: string;
  imagen?: string;
}

interface Programas {
  titulo?: string;
  descripcion?: string;
  url?: string;
  descarga?: string;
}

type TotalDescargas = Record<string, InfoDescarga>;

export interface InfoDescarga {
  identificador: string;
  nombre: string;
  archivo: string;
  descarga: string;
  progreso: string;
  finalizado: boolean;
}

File: src/podcast/modelos.rs
use serde::Serialize;

#[derive(Serialize)]
pub struct Podcast {
    pub error_critico: Option<String>,
    pub errores: Vec<String>,
    pub info: Info,
    pub programas: Vec<Programa>,
}

#[derive(Serialize)]
pub struct Info {
    pub nombre: Option<String>,
    pub descripcion: Option<String>,
    pub imagen: Option<String>,
}

#[derive(Serialize)]
pub struct Programa {
    pub titulo: Option<String>,
    pub descripcion: Option<String>,
    pub url: Option<String>,
    pub descarga: Option<String>,
}
